hail/src/main/scala/is/hail/utils/package.scala

```diff
@@ -595,13 +595,11 @@ package object utils extends Logging
     "part-" + StringUtils.leftPad(is, numDigits, "0")
   }
 
-  def partSuffix(ctx: TaskContext): String = {
-    val rng = new java.security.SecureRandom()
-    val fileUUID = new java.util.UUID(rng.nextLong(), rng.nextLong())
-    s"${ ctx.stageId() }-${ ctx.partitionId() }-${ ctx.attemptNumber() }-$fileUUID"
-  }
+  def partSuffix(ctx: TaskContext): String =
+    s"${ ctx.stageId() }-${ ctx.partitionId() }-${ ctx.attemptNumber() }"
 
-  def partFile(d: Int, i: Int, ctx: TaskContext): String = s"${ partFile(d, i) }-${ partSuffix(ctx) }"
+  def partFile(d: Int, i: Int, ctx: TaskContext): String =
+    s"${ java.util.UUID.randomUUID() }-${ partFile(d, i) }-${ partSuffix(ctx) }"
```

Comment:
I believe this complies with Google's suggestions. I assume we keep a list of the files to read somewhere else so this is forwards and backwards compatible?

Reply:
We do, our metadata files keep a list of partitions. The first partition is `partFile[0]`, second is `partFile[1]`, etc.. In cases where parallel we expect the output to be consumed by tools other than hail (parallel vcf or tsv/csv export) we use spark's facilities which produce `part-0nnnn` names.

Reply:
Hmm. Well, that's just another reason to stop using Spark!